runtime/onert/backend/xnnpack/ops/DepthwiseConvolutionLayer.cc

```diff
@@ -0,0 +1,149 @@
+                 const uint32_t stride_height, const uint32_t multiplier, const uint32_t dilation_width_factor,
+                 const uint32_t dilation_height_factor, const ir::Activation activation,
+                 IPortableTensor *output)
+{
+  _input = input;
+  _kernel = kernel;
+  _bias = bias;
+  _padding_type = padding_type;
+  _padding_left = padding_left;
+  _padding_right = padding_right;
+  _padding_top = padding_top;
+  _padding_bottom = padding_bottom;
+  _stride_width = stride_width;
+  _stride_height = stride_height;
+  _multiplier = multiplier;
+  _dilation_width_factor = dilation_width_factor;
+  _dilation_height_factor = dilation_height_factor;
+  _activation = activation;
+  _output = output;
+
+  // TODO Support not nhwc layer
+  assert(_input->layout() == ir::Layout::NHWC);
+
+  assert(_activation == ir::Activation::NONE || _activation == ir::Activation::RELU || 
+         _activation == ir::Activation::RELU1 || _activation == ir::Activation::RELU6);
+}
+
+void DepthwiseConvolutionLayer::run()
+{
+  prepare();
+
+  if (_input->data_type() == OperandType::FLOAT32)
+  {
+    enum xnn_status status = xnn_run_operator(_kernel_op, nullptr /* thread pool */);
+    if (status != xnn_status_success) {
+      throw std::runtime_error{"failed to run FP32 DepthwiseConvolution operator"};
+    }
+  }
+  else
+  {
+    throw std::runtime_error{"XNNPACK DepthwiseConv: unsupported data type"};
+  }
+}
+
+void DepthwiseConvolutionLayer::prepare()
+{
+  if (_prepare)
+    return;
+
+  float output_activation_min = 0.f, output_activation_max = 0.f;
+  CalculateActivationRange<float>(_activation, &output_activation_min, &output_activation_max);
+
+  // NHWC
+  // Kernel format is [1, kernel_height, kernel_width, depth_out].
+  const auto &kernel_shape = _kernel->getShape();
+  uint32_t kernel_height = kernel_shape.dim(1);
+  uint32_t kernel_width = kernel_shape.dim(2);
+  uint32_t output_channels = kernel_shape.dim(3);
+  uint32_t input_channels = _input->getShape().dim(3);
+  uint32_t input_width  = _input->getShape().dim(2);
+  uint32_t input_height  = _input->getShape().dim(1);
+  uint32_t batch_size = _input->getShape().dim(0);
+  assert(static_cast<uint32_t>(_output->getShape().dim(3)) == output_channels);
+  assert(output_channels == input_channels * _multiplier);
+
+  enum xnn_status status = xnn_create_convolution2d_nhwc_f32(
+      _padding_top, _padding_right, _padding_bottom, _padding_left,
+      kernel_height, kernel_width, _stride_height, _stride_width,
+      _dilation_height_factor, _dilation_width_factor,
+      input_channels /* groups */,
+      1 /* group_input_channels */,
+      _multiplier /* group_output_channels */,
+      input_channels /* input_channel_stride */,
+      output_channels /* output_channel_stride */,
+      reinterpret_cast<const float*>(_kernel->buffer()),
+      reinterpret_cast<const float*>(_bias->buffer()),
+      output_activation_min, output_activation_max,
+      0 /* flags */, &_kernel_op);
```

Comment:
@YongseopKim `XNN_FLAG_DEPTHWISE_CONVOLUTION` flag might be needed at here. https://github.com/google/XNNPACK/blob/9cef5eabcb5cc408db468abc0009dc424a852272/src/runtime.c#L354